packages/fast-components-react-msft/src/pivot/pivot.tsx

```diff
@@ -0,0 +1,287 @@
+            return {
+                activeId: nextProps.activeId,
+            };
+        }
+
+        return null;
+    }
+
+    protected handledProps: HandledProps<PivotHandledProps> = {
+        label: void 0,
+        items: void 0,
+        managedClasses: void 0,
+    };
+
+    private tabsRef: React.RefObject<any>;
+
+    private ltr: Direction;
+
+    private tabPanelIndex: number;
+
+    /**
+     * The constructor
+     */
+    constructor(props: PivotProps) {
+        super(props);
+
+        const activeId: string = !this.props.activeId
+            ? this.props.items
+                ? this.props.items[0].id
+                : ""
+            : this.props.activeId;
+
+        this.state = {
+            offsetX: 0,
+            activeId,
+            focused: false,
+        };
+
+        this.tabsRef = React.createRef();
+    }
+
+    public componentDidMount(): void {
+        this.ltr = this.getLTR();
+        this.setActiveIndicatorOffset();
+        this.tabPanelIndex = this.updateTabPanelIndex();
+    }
+
+    public componentDidUpdate(): void {
+        if (this.ltr !== this.getLTR()) {
+            this.setActiveIndicatorOffset();
+            this.ltr = this.getLTR();
+        }
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLButtonElement | HTMLAnchorElement> {
+        return (
+            <BaseTabs
+                {...this.unhandledProps()}
+                ref={this.tabsRef}
+                managedClasses={this.generatePivotClassNames()}
+                activeId={this.state.activeId}
+                onUpdate={this.handleTabsUpdate}
+                items={this.props.items}
+                label={this.props.label}
+                onBlur={this.handleOnBlur}
+                onFocus={this.handleOnFocus}
+            >
+                <span
+                    style={{ transform: `translateX(${toPx(this.state.offsetX)})` }}
+                    className={this.generateActiveIndicatorClassNames()}
+                />
+            </BaseTabs>
+        );
+    }
+
+    /**
+     * Returns tabs managedclasses with new carousel-specific JSS
+     */
+    protected generatePivotClassNames(): TabsClassNameContract {
+        return {
+            tabs: get(this.props, "managedClasses.pivot", ""),
+            tabs_tabPanels: this.generateTabPanelsClassNames(),
+            tabs_tabList: get(this.props, "managedClasses.pivot_itemList", ""),
+            tabs_tabPanelContent: get(
+                this.props,
+                "managedClasses.pivot_tabPanelContent",
+                ""
+            ),
+            tab: get(this.props, "managedClasses.pivot_item", ""),
+            tab__active: get(this.props, "managedClasses.pivot_item__active", ""),
+            tabPanel: get(this.props, "managedClasses.pivot_tabPanel", ""),
+            tabPanel__hidden: get(
+                this.props,
+                "managedClasses.pivot_tabPanel__hidden",

```

Comment:
nit: should probably format the get functions consistently in the block (either all wrap or not)

Reply:
That's prettier doing its thing. I might be able to line break the other lines but it may just format it back.

Reply:
I'd just let prettier do it's thing.